mlrun/runtimes/function.py

```diff
@@ -542,7 +559,7 @@ def deploy_nuclio_function(function: RemoteRuntime, dashboard="", watch=False):
         )
         update_in(config, "metadata.name", function.metadata.name)
         update_in(config, "spec.volumes", function.spec.to_nuclio_vol())
-        base_image = get_in(config, "spec.build.baseImage")
+        base_image = get_in(config, "spec.build.baseImage") or function.spec.image
```

Comment:
Why use the spec's image as the base image ? those are completely different things no ?

Reply:
Also, shouldn't you `enrich_image_url` here ?

Reply:
talked offline, the enrich is called in the next 2 lines